flopy/discretization/modeltime.py

```diff
@@ -7,61 +12,211 @@ class ModelTime:
 
     Parameters
     ----------
-    stress_periods : pandas dataframe
-        headings are: perlen, nstp, tsmult
-    temporal_reference : TemporalReference
-        contains start time and time units information
+    perlen : list, np.ndarray
+        list or numpy array of stress-period lengths
+    nstp : list, np.ndarray
+        list or numpy array of number of time-steps per stress period
+    tsmult : list, np.ndarray
+        list or numpy array of timestep mult infomation
+    time_units : int or str
+        string or pre-mf6 integer representation (ITMUNI) of time units
+    start_datetime : various objects
+        user supplied datetime representation. Please see the
+        ModelTime.datetime_from_user_input documentation for a list
+        of the supported representation types
+    steady_state : list, np.ndarray
+        optional list or numpy array of boolean flags that determine identify
+        steady-state or transient stress-periods
     """
 
     def __init__(
         self,
-        period_data=None,
-        time_units="days",
+        perlen,
+        nstp,
+        tsmult=None,
+        time_units=None,
         start_datetime=None,
         steady_state=None,
     ):
-        self._period_data = period_data
-        self._time_units = time_units
-        self._start_datetime = start_datetime
+        nrecs = len(perlen)
+        if tsmult is None:
+            tsmult = np.full((nrecs,), 1)
+
+        if nrecs != len(nstp):
+            raise AssertionError()
+
+        if len(tsmult) != len(nstp):
+            raise AssertionError
+
+        self._perlen = perlen
+        self._nstp = nstp
+        self._tsmult = tsmult
+        self._time_units = self.timeunits_from_user_input(time_units)
+        self._start_datetime = self.datetime_from_user_input(start_datetime)
         self._steady_state = steady_state
```

Comment:
I don't think it's necessary and it could be removed. I don't see a real use for it within the class

Reply:
Seems like this could be useful for converting simulation time to actual datetimes? For example, if the model starts with a steady-state period, that probably shouldn't be included in the elapsed time.